tt_metal/jit_build/data_format.cpp

```diff
@@ -264,25 +212,7 @@ std::vector<DataFormat> get_unpack_dst_formats(
         } else if (int_fpu_en) {
             unpack_dst_format.push_back(src_format);
         } else {
-            if (input_formats[i] == DataFormat::Float32 && !unpack_to_dest_mode.empty() && unpack_to_dest_mode[i] != UnpackToDestMode::Default) {
-                unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, DataFormat::Float32));
-            } else {
-                unpack_dst_format.push_back(get_single_unpack_dst_format(input_formats[i], pack_format, unpack_cond_dst_format));
-            }
```

Comment:
This if-else statement needs stay, it enables fp32 unpack to dest. Also, probably need to keep the lines related to `const bool en_unpack_tf32` a few lines up in this function.